apollo-router/src/plugins/telemetry/config_new/events.rs

```diff
@@ -664,9 +753,14 @@ mod tests {
                         .build()
                         .unwrap(),
                     |_r| {
+                        let context_with_error = Context::new();
```

Comment:
Let's separate this out into a different test, and also add tests for response headers.